Subject: [PATCH] flourite: fall back to standard drawable loading

Adapted from AdarshGrewal's resource-loader fix in rodin-miuicamera.
Use flourite's class names and AndroidX ResourcesCompat entry point.
Target: flourite Camera 6.4.000320.1 (OS3.0.304.0.WPRMIXM).

diff --git a/smali_classes4/F8/a.smali b/smali_classes4/F8/a.smali
--- a/smali_classes4/F8/a.smali
+++ b/smali_classes4/F8/a.smali
@@ -166,7 +166,7 @@
 .end method
 
 .method public static c(Landroid/content/res/Resources;ILjava/lang/Integer;)Landroid/graphics/drawable/Drawable;
-    .locals 5
+    .locals 6
     .param p1    # I
         .annotation build Landroidx/annotation/IdRes;
         .end annotation
@@ -232,6 +232,8 @@
     move p2, v2
 
     :goto_0
+    move v5, p1
+
     const/4 v3, 0x1
 
     invoke-virtual {p0, p1, p2, v0, v3}, Landroid/content/res/Resources;->getValueForDensity(IILandroid/util/TypedValue;Z)V
@@ -252,9 +254,12 @@
 
     move-result-object p2
 
+    :try_start_9
     invoke-virtual {p2, p1}, Landroid/content/res/AssetManager;->open(Ljava/lang/String;)Ljava/io/InputStream;
 
     move-result-object p2
+    :try_end_9
+    .catch Ljava/io/FileNotFoundException; {:try_start_9 .. :try_end_9} :catch_9
 
     :try_start_0
     invoke-static {p2}, Lkotlin/jvm/internal/l;->c(Ljava/lang/Object;)V
@@ -311,6 +316,17 @@
 
     goto :goto_3
 
+    :catch_9
+    move-exception p2
+
+    const/4 p2, 0x0
+
+    invoke-static {p0, v5, p2}, Landroidx/core/content/res/ResourcesCompat;->getDrawable(Landroid/content/res/Resources;ILandroid/content/res/Resources$Theme;)Landroid/graphics/drawable/Drawable;
+
+    move-result-object p0
+
+    return-object p0
+
     :goto_2
     :try_start_3
     throw p0
